lib/CArtHandler.cpp

```diff
@@ -740,15 +740,16 @@ void CArtHandler::erasePickedArt(const ArtifactID & id)
 {
 	CArtifact *art = objects[id];
 
-	if(auto artifactList = listFromClass(art->aClass))
+	if(auto artl = listFromClass(art->aClass))

```

Comment:
Why the change? Old code was fine with access via dereferences

Reply:
@IvanSavenko because there is std::reference_wrapper now here. We should get a reference from it somewhere. So, it is split to 2 methods - firstly, get an optional, and second - get actual reference from wrapper, so, I actually split this check to 2.